### PR TITLE
test:(getCompletedBoosts function)

### LINKED ISSUE #617: test: `/boost/get_completed_boosts?addr=${addr}`
read description here - https://github.com/starknet-id/starknet.quest/issues/602#issue-2255907921

### PR DESCRIPTION
**Pull request by mubarak23**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

Resolves: [#617](https://github.com/starknet-id/starknet.quest/issues/617)

### CODE REVIEW

File: tests/services/apiService.test.js [hunk 1]
```diff
@@ -1,42 +1,108 @@
...
+
+    const result = await getCompletedBoosts('string');
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/boost/get_completed_boosts?addr=string`
+    );
+
+    expect(result).toEqual(mockDataResponse);
+  });
+
+  it('should handle fetch with handle error gracefully', async () => {
```
Comment (ayushtom) [inline tests/services/apiService.test.js:1082]: i think you can handle this as a 500 error. The above test case handles this error too

Comment (mubarak23) [reply tests/services/apiService.test.js:1082]: @ayushtom do you mean, the i should check the status value in my test assertion

Comment (ayushtom) [reply tests/services/apiService.test.js:1082]: yes

Comment (mubarak23) [reply tests/services/apiService.test.js:1082]: Alright

**Comment by mubarak23** [comment]
@ayushtom check for 500 status done on new commit, kindly check

**Comment by mubarak23** [comment]
@ayushtom merge conflict fixed, kindly review

**Comment by ayushtom** [comment]
@mubarak23 i am not sure to understand why but your test cases seem to be failing, can you check once why this might be happening

**Comment by mubarak23** [comment]
currently looking at the test logs, i will find out why and fix it

**Comment by mubarak23** [comment]
@ayushtom kindly review and run the test on your end

**Comment by mubarak23** [comment]
anything i need to change here @ayushtom

**Comment by mubarak23** [comment]
am getting the following error when i run `npm run build ` @ayushtom 

  ```
Collecting page data  . ⚠ "next" should not be imported directly, imported in /Users/macbookpro/Documents/onlydust/starknet.quest/.next/server/app/categories/[category]/page.js
See more info here: https://nextjs.org/docs/messages/import-next
TypeError [ERR_INVALID_URL]: Invalid URL
```

**Comment by mubarak23** [comment]
Screenshot 2024-04-30 at 11 40 30 @ayushtom see the error on npm run build from local

**Comment by mubarak23** [comment]
am still getting this error when i ran `npm run build` @ayushtom 
Screenshot 2024-04-30 at 12 30 49

**Comment by mubarak23** [comment]
Screenshot 2024-04-30 at 13 32 01

**Comment by ayushtom** [comment]
@mubarak23 the bus error is i think not related to the repo
https://stackoverflow.com/questions/73022976/bus-error-when-running-npm-run-dev-npm-start

**Comment by mubarak23** [comment]
> @mubarak23 the bus error is i think not related to the repo https://stackoverflow.com/questions/73022976/bus-error-when-running-npm-run-dev-npm-start

i have fixed this issue already, it has to do with node_modules

**Comment by mubarak23** [comment]
> am still getting this error when i ran `npm run build` @ayushtom Screenshot 2024-04-30 at 12 30 49

@ayushtom see the error am getting

**Comment by mubarak23** [comment]
> Screenshot 2024-04-30 at 13 32 01

and this one @ayushtom

**Comment by mubarak23** [comment]
Screenshot 2024-04-30 at 18 37 17

@ayushtom  

```
Build error occurred
Error: Failed to collect page data for /achievements
```

**Comment by mubarak23** [comment]
Build error occurred
```
Error: Failed to collect page data for /404
    at /Users/macbookpro/Documents/onlydust/starknet.quest/node_modules/next/dist/build/utils.js:1268:15
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  type: 'Error'
}
```
   Collecting page data  .

@ayushtom

**Comment by ayushtom** [comment]
photo_2024-05-01_02-32-28

the build error i see which is not related to your error but a typing issue which you have in code

**Comment by mubarak23** [comment]
@ayushtom kindly resolve the conflict

**Comment by mubarak23** [comment]
> photo_2024-05-01_02-32-28
> 
> the build error i see which is not related to your error but a typing issue which you have in code

all my commit did not touch this file, so why is this happening, looking the code here i can see the error on that line

**Comment by mubarak23** [comment]
am going to close this and clone the repo again

**Pull request closed without merging**